homeassistant/components/synology_dsm/__init__.py

```diff
@@ -234,15 +242,17 @@ async def async_coordinator_update_data_cameras():
             }
         }
 
-    async def async_coordinator_update_data_central():
+    async def async_coordinator_update_data_central() -> None:
         """Fetch all device and sensor data from api."""
         try:
             await api.async_update()
         except Exception as err:
             raise UpdateFailed(f"Error communicating with API: {err}") from err
         return None
 
-    async def async_coordinator_update_data_switches():
+    async def async_coordinator_update_data_switches() -> dict[
+        str, dict[str, any]
```

Comment:
Have you tried to run mypy on this code?

Reply:
mypy is part of the CI checks

Reply:
The problem is right now mypy only checks strictly typed parts of HA. Everything else is ignored. I'm trying to change this in https://github.com/home-assistant/core/pull/49270

This code looks broken at least because of `any`.

Please don't add broken type hints and wait before https://github.com/home-assistant/core/pull/49270 is merged.

Reply:
Thanks for pointing that out ... I thought I already replaced all `any` by `Any`, but this two I have overseen :see_no_evil: